lib/eaco/adapters/active_record/compatibility.rb

```diff
@@ -41,7 +41,12 @@ def check!
         # @return [ActiveRecord::Base] associated with the model
         #
         def target
-          @model.base_class.superclass
+          target = @model.base_class.superclass
+          if target == ApplicationRecord
+            target.superclass
+          else
+            target
+          end
```

Comment:
Do we even need this here. At this point we already know we are plugging into ActiveRecord, so why can't we just use ActiveRecord::Base directly?